generators/java/templates/libraries/jersey2/oneof_model.mustache

```diff
@@ -0,0 +1,391 @@
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import {{invokerPackage}}.JSON;
+
+{{>additionalModelTypeAnnotations}}{{>generatedAnnotation}}{{>xmlAnnotation}}
+@JsonDeserialize(using = {{classname}}.{{classname}}Deserializer.class)
+@JsonSerialize(using = {{classname}}.{{classname}}Serializer.class)
+public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-implements}} implements {{{.}}}{{^-last}}, {{/-last}}{{/vendorExtensions.x-implements}} {
+    private static final Logger log = Logger.getLogger({{classname}}.class.getName());
+
+    public static class {{classname}}Serializer extends StdSerializer<{{classname}}> {
+        public {{classname}}Serializer(Class<{{classname}}> t) {
+            super(t);
+        }
+
+        public {{classname}}Serializer() {
+            this(null);
+        }
+
+        @Override
+        public void serialize({{classname}} value, JsonGenerator jgen, SerializerProvider provider) throws IOException, JsonProcessingException {
+            jgen.writeObject(value.getActualInstance());
+        }
+    }
+
+    public static class {{classname}}Deserializer extends StdDeserializer<{{classname}}> {
+        public {{classname}}Deserializer() {
+            this({{classname}}.class);
+        }
+
+        public {{classname}}Deserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public {{classname}} deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+            JsonNode tree = jp.readValueAsTree();
+            Object deserialized = null;
+            {{#useOneOfDiscriminatorLookup}}
+            {{#discriminator}}
+            {{classname}} new{{classname}} = new {{classname}}();
+
+            Map<String, Object> result2 = tree.traverse(jp.getCodec()).readValueAs(new TypeReference<Map<String, Object>>() {});
+
+            {{! Added code based on com/onfido/model/IdentityEnhancedReport.java - BEGIN }}
+            new{{classname}}.id = UUID.fromString((String)result2.get(JSON_PROPERTY_ID));
+            new{{classname}}.createdAt = OffsetDateTime.parse((String)result2.get(JSON_PROPERTY_CREATED_AT));
+            new{{classname}}.href = (String)result2.get(JSON_PROPERTY_HREF);
+            new{{classname}}.status = ReportStatus.fromValue((String)result2.get(JSON_PROPERTY_STATUS));
+            new{{classname}}.result = ReportResult.fromValue((String)result2.get(JSON_PROPERTY_RESULT));
+            new{{classname}}.subResult = ReportSubResult.fromValue((String)result2.get(JSON_PROPERTY_SUB_RESULT));
+            new{{classname}}.checkId = UUID.fromString((String)result2.get(JSON_PROPERTY_CHECK_ID));
+            new{{classname}}.documents = (List<ReportDocument>)(List<?>) result2.get(JSON_PROPERTY_DOCUMENTS);
```

Comment:
These lines of code should be improved to not access directly the HashMap but using the ReportShared model instead to take advantage from all the validation already implemented.